crates/bevy_pbr/src/light.rs

```diff
@@ -52,6 +53,84 @@ impl PointLightUniform {
     }
 }
 
+/// A Directional light.
+///
+/// Directional lights don't exist in reality but they are a good
+/// approximation for light sources VERY far away, like the sun or
+/// the moon.
+#[derive(Debug, Reflect)]
+#[reflect(Component)]
+pub struct DirectionalLight {
+    pub color: Color,
+    pub intensity: f32,
+    direction: Vec3,
+}
+
+impl DirectionalLight {
+    /// Create a new directional light component.
+    ///
+    /// # Panics
+    /// Will panic if `direction` is not normalized.
+    pub fn new(color: Color, intensity: f32, direction: Vec3) -> Self {
+        assert!((direction.length_squared() - 1.0).abs() < 0.0001);
```

Comment:
You should provide an assert message with this. 
And also we should add `#[track_caller]` if this can panic.
e.g.
```rust
assert!(condition, "message");
```

Alternative:
We could have this return an `Option<Self>`  and then have the caller manually `unwrap` the value.

Reply:
I read the explanation in the [reference ](https://doc.rust-lang.org/reference/attributes/codegen.html) but I found it super confusing. What would be the goal of adding `#[track_caller]` here?

Reply:
> I read the explanation in the [reference ](https://doc.rust-lang.org/reference/attributes/codegen.html) but I found it super confusing. What would be the goal of adding `#[track_caller]` here?

When someone calls `DirectionalLight::new` with an invalid direction, by default the panic message will be something like
``thread 'main' panicked at 'assetion failed: <the failed expr>', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_pbr/src/light.rs:75:10``, from which you can't tell where the panic originates from unless you enable the backtrace.
When adding the `#[track_caller]` attribute, the reported path will be `src/main.rs:20:2`, so you instantly now where you caused the panic.